src/components/schedule-page/table/schedule/schedule-parts/shift-cell/shift-cell.component.tsx

```diff
@@ -100,73 +104,97 @@ export function ShiftCellComponentF(options: ShiftCellOptions): JSX.Element {
     ),
     [isBlocked, onClick]
   );
+  let clearModal;
 
-  //  #region view
+  //  #region view?
   return (
-    <td
-      ref={mergeRefs([selectableItemRef, componentContainer])}
-      className={classNames("mainCell", {
-        selection: isSelected || (isComponentVisible && isBlocked),
-        blocked: isBlocked,
-      })}
-      onClick={(): void => toggleComponentVisibility()}
-      id={id}
-      onBlur={(): void => {
-        onBlur?.();
-      }}
-    >
-      {showInput && (
-        <WrapContentDiv>
-          <CellInput
-            input={ShiftAutocompleteComponent}
-            onKeyDown={onKeyDown}
-            onValueChange={_onValueChange}
-            {...options}
-          />
-        </WrapContentDiv>
+    <>
+      {showInput && isComponentVisible && (
+        <div
+          style={{
+            position: "absolute",
+            top: 0,
+            left: 0,
+            width: "100vw",
+            height: "100vh",
+            zIndex: 100,
+          }}
+          onClick={(): void => setIsComponentVisible(false)}
+        />
       )}
-      <ErrorTooltipProvider
-        className="wrapContent"
-        errorSelector={errorSelector}
-        showTooltip={!showInput}
+      <div
+        ref={mergeRefs([selectableItemRef, componentContainer])}
+        className={classNames("mainCell", {
+          selection: isSelected || (isComponentVisible && isBlocked),
+          blocked: isBlocked,
+        })}
+        onClick={(): void => toggleComponentVisibility()}
+        onMouseEnter={(): void => {
+          clearTimeout(clearModal);
+        }}
+        onMouseLeave={(): void => {
+          clearModal = setTimeout(() => {
+            setIsComponentVisible(false);
```

Comment:
Podobnie jak poprzendnio. Coś mi tutaj śmierdzi wyciekiem pamięci. Dodatkowo mamy powtórzenie kodu. Wydaje mi się, że gdybyś to wyniósł do useEffecta to można by to umieścić w osobny pliku i zrobić hooka, którego będziemy sobie mogli reużywać.